src/main/java/parser/CommentParser.java

```diff
@@ -0,0 +1,32 @@
+package parser;
+
+import command.Command;
+import command.CommentCommand;
+import exception.DukeException;
+
+
+
+public class CommentParser extends IndexParser {
+
+    public CommentParser(String userInput, String command) {
+        super(userInput, command);
+    }
+
+    @Override
+    public Command parse() throws DukeException {
+        super.extract();
+        String comment = extractComment(taskFeatures);
+        return new CommentCommand(indexOfTask, comment);
+    }
+
+    private String extractComment(String taskFeatures) throws DukeException {
+        String comment;
+        try {
+            String[] commentCommandParts = taskFeatures.split(" ", 2);
```

Comment:
Minor remark but when splitting strings, preferable to use regex "\\s+". The regex detects whitespace characters, even multiple spaces typed in consequently, as a single unit. " " detects only one space and then executes the split on that sole space.
The latter problem can be mitigated with trim() though.